src/app/accounts/get-accounts-by-custom-fields.ts

```diff
@@ -0,0 +1,26 @@
+import { getAccountsConfig } from "@config"
+
+import { InvalidCustomFieldError } from "@domain/accounts"
+
+import { AccountCustomFieldsRepository, AccountsRepository } from "@services/mongoose"
+
+const { customFields: customFieldsSchema } = getAccountsConfig()
+
+export const getAccountsByCustomFields = async ({
+  field,
+  value,
+}: {
+  field: string
+  value: string
+}): Promise<Account[] | ApplicationError> => {
+  const fieldSchema = customFieldsSchema.find((s) => s.name === field)
+  if (!fieldSchema) return new InvalidCustomFieldError()
+
+  const accountIds = await AccountCustomFieldsRepository().listAccountIdsByCustomField({
```

Comment:
Up to now we have favoured the style of `listByCustomField` and then filtered `.map(|f| f.account_id)` in the domain.
This here is encoding use case into a repo which should purely be for data access.

Reply:
Updated with the new model https://github.com/GaloyMoney/galoy/pull/1465/commits/22c9f29082861a0a3ae06bfd1c0c82247e53368b